gnat2goto/ireps/ireps_generator.py

```diff
@@ -205,2067 +205,2071 @@ def mk_prefixed_lines(prefix, lines, join=""):
+
+        for fld_kind, all_acc in accessors.iteritems():
+            n = req_fld[fld_kind]
+            f = new_file("%s_%u.smt2" % (fld_kind, n))
+
+            write(f, "(set-logic QF_LIA)")
+            write(f, "(set-option :produce-models true)")
+
+            # Accessors
+            for acc in all_acc:
+                write(f, "(declare-const acc_%s Int)" % acc)
+
+            # Restrict to fields
+            for acc in all_acc:
+                write(f, "(assert (<= 0 acc_%s %u))" % (acc, n - 1))
+
+            # required accessors must be disjoint
+            for sn in layout:
+                tmp = []
+                for friendly_name in layout[sn]:
+                    if friendly_name in all_acc:
+                        tmp.append("acc_" + friendly_name)
+                if len(tmp) >= 2:
+                    write(f, ";; for %s" % sn)
+                    write_raw(f, "(assert (distinct %s))" % " ".join(tmp))
+
+            write(f, "(check-sat)")
+            write(f, "(get-model)")
+            write(f, "(exit)")
+            write_file(f)
+
+            os.system("cvc4 %s_%u.smt2 > %s_%u.out" % (fld_kind, n,
+                                                    fld_kind, n))
+
+            with open("%s_%u.out" % (fld_kind, n), "rU") as fd:
+                tmp = fd.read().strip()
+            assert tmp.startswith("sat")
+
+            optimal_lo = {}
+            for line in tmp.splitlines():
+                if line.startswith("(define-fun acc_"):
+                    _, nam, _, _, pos = line.split()
+                    nam = nam.replace("acc_", "")
+                    assert nam in all_acc
+                    pos = int(pos.rstrip(")"))
+                    assert 0 <= pos < n
+                    optimal_lo[nam] = pos
+            assert sorted(optimal_lo) == all_acc
+
+            # lo ::= schema -> friendly_name -> (str|int|bool|sloc,
+            #                                    index,
+            #                                    irep|list|trivial)
+
+            for sn in layout:
+                for friendly_name in layout[sn]:
+                    if friendly_name in optimal_lo:
+                        lo_typ, lo_idx, lo_kind = layout[sn][friendly_name]
+                        lo_idx = optimal_lo[friendly_name]
+                        layout[sn][friendly_name] = (lo_typ, lo_idx, lo_kind)
+
+
+        return layout, req_fld["int"], req_fld["bool"]
+
+    def generate_code(self, optimize, schema_file_names):
+        schemata = {}
+        for schema_fn in schema_file_names:
+            with open(schema_fn, "rU") as fd:
+                sn = os.path.splitext(os.path.basename(schema_fn))[0]
+                # print "Loading %s" % sn
+                schemata[sn] = json.load(fd)
+                schemata[sn]["subclasses"] = set()
+                schemata[sn]["ada_name"] = ada_casing("i_" + sn)
+
+        # schemata will contain the following. Things in <> we synthesise below.
```

Comment:
Why was this changed from upper to lower case?

Reply:
Accidentally. Will be fixed soon.